File: src/com/wasteinformationserver/website/basicrequest/PostRequest.java
package com.wasteinformationserver.website.basicrequest;

import com.sun.net.httpserver.Headers;
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.util.HashMap;

public abstract class PostRequest implements HttpHandler {

    @Override
    public void handle(HttpExchange httpExchange) throws IOException {
        if (httpExchange.getRequestMethod().equals("POST")) {
            StringBuilder sb = new StringBuilder();
            InputStream ios = httpExchange.getRequestBody();
            int i;
            while ((i = ios.read()) != -1) {
                sb.append((char) i);
            }
            String query =  sb.toString();

            HashMap<String, String> params = new HashMap<>();

            String[] res = query.split("&");
            for (String str : res) {
                String[] values = str.split("=");
                params.put(values[0], values[1]);
            }

            String response = request(params);


            Headers h = httpExchange.getResponseHeaders();
            h.set("Content-Type", "application/json");
            httpExchange.sendResponseHeaders(200, 0);

            OutputStream os = httpExchange.getResponseBody();
            os.write(response.getBytes());
            os.close();
        }
    }

    public abstract String request(HashMap<String, String> params);

}

File: src/java/com/wasteinformationserver/website/basicrequest/PostRequest.kt
package com.wasteinformationserver.website.basicrequest

import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import java.io.IOException
import java.util.*

/**
 * basic POST request handler
 * reply function has to be implemented!
 */
abstract class PostRequest : HttpHandler {
    @Throws(IOException::class)
    override fun handle(httpExchange: HttpExchange) {
        if (httpExchange.requestMethod == "POST") {
            val sb = StringBuilder()
            val ios = httpExchange.requestBody
            var i: Int
            while (ios.read().also { i = it } != -1) {
                sb.append(i.toChar())
            }
            val query = sb.toString()
            val params = HashMap<String, String>()
            val res = query.split("&".toRegex()).toTypedArray()
            for (str in res) {
                val values = str.split("=".toRegex()).toTypedArray()
                params[values[0]] = values[1]
            }
            val response = request(params)
            val h = httpExchange.responseHeaders
            h["Content-Type"] = "application/json"
            httpExchange.sendResponseHeaders(200, 0)
            val os = httpExchange.responseBody
            os.write(response.toByteArray())
            os.close()
        }
    }

    /**
     * @param params received get params from com.wasteinformationserver.website
     * @return json reply to com.wasteinformationserver.website
     */
    abstract fun request(params: HashMap<String, String>?): String
}